examples/inverse/plot_rap_music.py

```
@@ -0,0 +1,110 @@
+from mne.beamformer import rap_music
+from mne.viz import plot_dipoles
+
+data_path = sample.data_path()
+subjects_dir = data_path + '/subjects'
+fwd_fname = data_path + '/MEG/sample/sample_audvis-meg-eeg-oct-6-fwd.fif'
+evoked_fname = data_path + '/MEG/sample/sample_audvis-ave.fif'
+cov_fname = data_path + '/MEG/sample/sample_audvis-cov.fif'
+
+# Read the evoked response and crop it
+condition = 'Left Auditory'
+evoked = mne.read_evokeds(evoked_fname, condition=condition,
+                          baseline=(None, 0))
+evoked.crop(tmin=-50e-3, tmax=300e-3)
+
+# Set up pick list: EEG + MEG - bad channels (modify to your needs)
+# left_temporal_channels = mne.read_selection('Right-temporal')
+# left_temporal_channels = mne.read_selection('Right-temporal')
+# evoked = mne.pick_types_evoked(evoked, meg=False, eeg=False, include=left_temporal_channels)
+
+evoked = mne.pick_types_evoked(evoked, meg=True, eeg=False)
+
+# Read the forward solution
+forward = mne.read_forward_solution(fwd_fname, surf_ori=True,
+                                    force_fixed=False)
+
+# Read noise covariance matrix and regularize it
+noise_cov = mne.read_cov(cov_fname)
+
+dipoles, residual = rap_music(evoked, forward, noise_cov, n_dipoles=2,
+                              return_residual=True, verbose=True,
+                              signal_ndim=2)
+coord_trans = forward['mri_head_t']['trans']
+plot_dipoles(dipoles, coord_trans, subject='sample',
+             subjects_dir=subjects_dir, colors=[(0., 0., 1.), (1., 0., 0.)])
+
+# Plot the time-series
+plt.figure()
+plt.plot(dipoles[0].times, dipoles[0].amplitude, 'b', linewidth=1.5)
+plt.plot(dipoles[1].times, dipoles[1].amplitude, 'r', linewidth=1.5)
+plt.ylabel('amplitude (nAm)')
+plt.xlabel('times (ms)')
+
+# Plot the evoked data and the residual.
+evoked.plot(ylim=dict(grad=[-300, 300], mag=[-800, 800]))
+residual.plot(ylim=dict(grad=[-300, 300], mag=[-800, 800]))
+
+
+
+
+# ############### test ########
+import numpy as np
```

this should throw a pep8 error with the latest version, all imports should be at the beginning of the file.